contracts/vesting/src/traits/vesting_account.rs

```diff
@@ -0,0 +1,65 @@
+use crate::errors::ContractError;
+use cosmwasm_std::{Coin, Env, Storage, Timestamp};
+
+pub trait VestingAccount {
+    // locked_coins returns the set of coins that are not spendable (can still be delegated tough) (i.e. locked),
+    // defined as the vesting coins that are not delegated.
+    //
+    // To get spendable coins of a vesting account, first the total balance must
+    // be retrieved and the locked tokens can be subtracted from the total balance.
+    // Note, the spendable balance can be negative.
```

Comment:
It can be negative if more tokens are delegated/bonded or vesting then there are vested tokens, so when calculating it we go with the saturating sub, as we don't really care if its negative or zero, we just want to block a withdraw